test/cases/filesystem.sh

```diff
@@ -191,9 +192,271 @@ size = 131072000
 mountpoint = "/foobar"
 size = 131072000
 EOF
+    EXPECTED_MOUNTPOINTS=(
+        "/"
+        "/var"
+        "/var/log"
+        "/var/log/audit"
+        "/usr"
+        "/tmp"
+        "/var/tmp"
+        "/home"
+        "/home/shadowman"
+        "/opt"
+        "/srv"
+        "/app"
+        "/data"
+        "/boot"
+        "/boot/firmware"
+        "/foobar"
+    )
+}
+
+write_plain_blueprint() {
+    tee "$BLUEPRINT_FILE" > /dev/null << EOF
+name = "custom-filesystem"
+description = "A base system with custom plain partitions"
+version = "0.0.1"
+
+[[customizations.disk.partitions]]
+mountpoint = "/data"
+fs_type = "ext4"
+minsize = "1 GiB"
+
+[[customizations.disk.partitions]]
+mountpoint = "/home"
+label = "home"
+fs_type = "ext4"
+minsize = "2 GiB"
+
+[[customizations.disk.partitions]]
+mountpoint = "/home/shadowman"
+fs_type = "ext4"
+minsize = "500 MiB"
+
+[[customizations.disk.partitions]]
+mountpoint = "/foo"
+fs_type = "ext4"
+minsize = "1 GiB"
+
+[[customizations.disk.partitions]]
+mountpoint = "/var"
+fs_type = "xfs"
+minsize = "4 GiB"
+
+[[customizations.disk.partitions]]
+mountpoint = "/opt"
+fs_type = "ext4"
+minsize = "1 GiB"
+
+[[customizations.disk.partitions]]
+mountpoint = "/media"
+fs_type = "ext4"
+minsize = "1 GiB"
+
+[[customizations.disk.partitions]]
+mountpoint = "/root"
+fs_type = "ext4"
+minsize = "1 GiB"
+
+[[customizations.disk.partitions]]
+mountpoint = "/srv"
+fs_type = "xfs"
+minsize = "1 GiB"
+
+[[customizations.disk.partitions]]
+fs_type = "swap"
+minsize = "1 GiB"
+EOF
+    EXPECTED_MOUNTPOINTS=(
+        "/data"
+        "/home"
+        "/home/shadowman"
+        "/foo"
+        "/var"
+        "/opt"
+        "/media"
+        "/root"
+        "/srv"
+        "swap"
+    )
+}
+
+write_lvm_blueprint() {
+    tee "$BLUEPRINT_FILE" > /dev/null << EOF
+name = "custom-filesystem"
+description = "A base system with custom LVM partitioning"
+
+[customizations.disk]
+type = "gpt"
+
+  [[customizations.disk.partitions]]
+  mountpoint = "/data"
+  minsize = "1 GiB"
+  label = "data"
+  fs_type = "ext4"
+
+  [[customizations.disk.partitions]]
+  type = "lvm"
+  name = "testvg"
+  minsize = 10_737_418_240
+
+    [[customizations.disk.partitions.logical_volumes]]
+    name = "homelv"
+    mountpoint = "/home"
+    label = "home"
+    fs_type = "ext4"
+    minsize = "2 GiB"
```

Comment:
Is this indentation semantically significant? I've never seen this in TOML :innocent:

Reply:
It's not.  I originally wrote this as json and converted to toml to make sure I did it right (those nested arrays can get weird) and the converter I used added the indentation and I kept it.  I think it makes it easier to read.

Reply:
I agree. And I thought that TOML can't get any more "obvious" :sweat_smile: